File: azor-the-chatdog-js/src/cli/console.ts
/**
 * Terminal output utilities
 * Color scheme matches Python version using colorama
 */

import chalk from 'chalk';

/**
 * Print error message in red (Fore.RED)
 */
export function printError(message: string): void {
  console.log(chalk.red(message));
}

/**
 * Print success message in green
 */
export function printSuccess(message: string): void {
  console.log(chalk.green(message));
}

/**
 * Print info message in default color (no color in Python version)
 */
export function printInfo(message: string): void {
  console.log(message);
}

/**
 * Print help message in yellow (Fore.YELLOW)
 */
export function printHelp(message: string): void {
  console.log(chalk.yellow(message));
}

/**
 * Print warning message in yellow
 */
export function printWarning(message: string): void {
  console.log(chalk.yellow(message));
}

/**
 * Print assistant message in cyan (Fore.CYAN)
 */
export function printAssistant(message: string): void {
  console.log(chalk.cyan(message));
}

/**
 * Print user message in blue (Fore.BLUE)
 */
export function printUser(message: string): void {
  console.log(chalk.blue(message));
}

/**
 * Display help information (matches Python version)
 */
export function displayHelp(sessionId?: string): void {
  if (sessionId) {
    printInfo(`Current session (ID): ${sessionId}`);
  }
  printHelp('Available commands (slash commands):');
  printHelp('  /switch <ID>      - Switch to an existing session.');
  printHelp('  /help             - Show this help.');
  printHelp('  /exit, /quit      - End the chat.');
  printHelp('\n  /session list     - List saved sessions.');
  printHelp('  /session display  - Show full session history.');
  printHelp('  /session pop      - Remove the last user/assistant message pair.');
  printHelp('  /session clear    - Clear the current session history.');
  printHelp('  /session new      - Start a new session.');
}

/**
 * Print a separator line
 */
export function printSeparator(): void {
  console.log(chalk.gray('─'.repeat(60)));
}

/**
 * Clear the console
 */
export function clearConsole(): void {
  console.clear();
}

File: azor-the-chatdog-py/src/cli/console.py
"""
Console output utilities for the chatbot.
Centralizes colorama usage for consistent terminal output.
"""
import sys
from colorama import init, Fore, Style
from files.config import LOG_DIR

init(autoreset=True)


def print_error(message: str):
    """Print an error message in red color.
    
    Args:
        message: The error message to display
    """
    print(Fore.RED + message + Style.RESET_ALL)


def print_assistant(message: str):
    """Print an assistant message in cyan color.
    
    Args:
        message: The assistant message to display
    """
    print(Fore.CYAN + message + Style.RESET_ALL)


def print_user(message: str):
    """Print a user message in blue color.
    
    Args:
        message: The user message to display
    """
    print(Fore.BLUE + message + Style.RESET_ALL)


def print_info(message: str):
    """Print an informational message in yellow color.
    
    Args:
        message: The informational message to display
    """
    print(message)

def print_help(message: str):
    """Print an informational message in yellow color.
    
    Args:
        message: The informational message to display
    """
    print(Fore.YELLOW + message + Style.RESET_ALL)


def display_help(session_id: str):
    """Displays a short help message."""
    print_info(f"Current session (ID): {session_id}")
    print_info(f"Session files are saved continuously in: {LOG_DIR}")
    print_help("Available commands (slash commands):")
    print_help("  /switch <ID>      - Switch to an existing session.")
    print_help("  /help             - Show this help.")
    print_help("  /exit, /quit      - End the chat.")
    print_help("\n  /session list     - List saved sessions.")
    print_help("  /session display  - Show full session history.")
    print_help("  /session pop      - Remove the last user/assistant message pair.")
    print_help("  /session clear    - Clear the current session history.")
    print_help("  /session new      - Start a new session.")


def display_final_instructions(session_id: str):
    """Displays instructions for continuing the session."""
    print_info("\n--- Resuming a session ---")
    print_info(f"To continue this session (ID: {session_id}) later, run:")
    print(Fore.WHITE + Style.BRIGHT + f"\n    python {sys.argv[0]} --session-id={session_id}\n" + Style.RESET_ALL)
    print("--------------------------------------\n")

